migrations/20170115_api_date_validation.js

```diff
@@ -0,0 +1,67 @@
+var migrationName = '20161230_nye_hats.js';
+var authorName = 'sprusr'; // in case script author needs to know when their ...
+var authorUuid = 'db711097-c5d0-4d25-8e5e-d955da4b79fd'; //... own data is done
+
+/*
+ * API date validation
+ */
+
+var mongo = require('mongoskin');
+
+var connectionString = 'mongodb://localhost:27017/habitrpg?auto_reconnect=true'; // FOR TEST DATABASE
+
+var dbTasks = mongo.db(connectionString).collection('tasks');
+
+var query = {
+  $nor: [
+    { date: { $type: "date" } },
```

Comment:
I don't think this is going to work as $nor matches docs whose date type is not a date AND where the date fields doesn't exist. And i don't think is necessary as all date fields are going to be strings if they exist. You can simply match `date: {$exists: true}`